apis/v1beta1/config.go

```diff
@@ -476,17 +489,27 @@ func (s *Service) ApplyDefaults(logger logr.Logger) error {
 
 	if tel.Metrics.Address != "" || len(tel.Metrics.Readers) != 0 {
 		// The user already set the address or the readers, so we don't need to do anything
+		logger.V(1).Info("telemetry configuration already provided by user, skipping defaults",
+			"metricsAddress", tel.Metrics.Address,
+			"readersCount", len(tel.Metrics.Readers))
 		return nil
 	}
 
+	logger.Info("no telemetry readers configuration found, applying default Prometheus endpoint")
+
 	host, port, err := s.MetricsEndpoint(logger)
 	if err != nil {
+		logger.Error(err, "failed to determine metrics endpoint for default configuration")
 		return err
 	}
 
 	reader := AddPrometheusMetricsEndpoint(host, port)
 	tel.Metrics.Readers = append(tel.Metrics.Readers, reader)
 
+	logger.Info("applied default Prometheus telemetry configuration",
```

Comment:
I think this message should not be in the default verbosity level.

Reply:
I don't agree 100% because most of the OTEL Config validations are in the default level, like when you forget to configure a receiver properly. So, I'd leave in the Info one.

Reply:
As we agreed through slack, I will create an event instead.